superset-frontend/spec/javascripts/datasource/DatasourceEditor_spec.jsx

```diff
@@ -85,11 +74,65 @@ describe('DatasourceEditor', () => {
     });
   });
 
-  it('merges columns', () => {
+  it('to add, remove and modify columns accordingly', () => {
+    const columns = [
+      {
+        name: 'ds',
+        type: 'DATETIME',
+        nullable: true,
+        default: '',
+        primary_key: false,
+      },
+      {
+        name: 'gender',
+        type: 'VARCHAR(32)',
+        nullable: true,
+        default: '',
+        primary_key: false,
+      },
+      {
+        name: 'new_column',
+        type: 'VARCHAR(10)',
+        nullable: true,
+        default: '',
+        primary_key: false,
+      },
+    ];
```

Comment:
This is the correct schema coming back from the API (compare with `extraColumn` above).